src/mpid/ch4/generic/mpidig_msg.h

```diff
@@ -42,7 +49,17 @@ MPL_STATIC_INLINE_PREFIX void MPIDIG_recv_copy(void *in_data, MPIR_Request * rre
     MPIDIG_rreq_async_t *p = &(MPIDIG_REQUEST(rreq, req->async));
     int is_contig = p->is_contig;
     MPI_Aint in_data_sz = p->in_data_sz;
-    if (is_contig && !p->iov_one.iov_len) {
+    if (is_contig == -1) {
```

Comment:
Using `MPIR_Typerep_unpack` is excellent, IMO. This use of `-1` though is making me a little queasy.

Reply:
I understand. I can use an extra flag or enum for it.

Reply:
Pushed last commit to define and use a proper enum type. 
In stead of `is_config` with 3 states, define `recv_type` with three enum values with proper names.